[gekjunxu] iP

gekjunxu opened the pull request:
# Bob Bot 🤖 

> **Bob** the Builder: Can We Fix it? Yes We Can! - **Bob** the Builder

Bob frees your mind of having to remember what you need to do. It is
- Text Based
- Java Based
- Bob Based
- **Freeeeeee~eee~**!! 😱 


To get started, simply:
1. Head to the Releases [page ](https://github.com/gekjunxu/ip/releases)to download the .jar file
2. Double click
3. Enjoy!


## Features 

- [x] Managing your todo list
- [x] Managing Deadlines
- [x] Keeping track of Events
- [x] Easily search your list for your wanted task


### Challenge yourself 🦾 
If you want to change the location of the data file, edit the `main` method in the Bob.java file:
```Java    
public static void main(String[] args) throws IOException, BobException {
        new Bob("data/bob.txt").run();
    }
```
Change the `data/bob.txt` part to your desired location. (Remember, use _relative_ paths!)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

gekjunxu committed:
Changed name to Bob

gekjunxu committed:
Changed name to Bob

gekjunxu committed:
Revert "Changed name to Bob"

This reverts commit 5e2c6604bd2fcc1388d57d502858c158232195fc.

gekjunxu committed:
Merge remote-tracking branch 'origin/master'

gekjunxu committed:
Merge branch 'master' of https://github.com/gekjunxu/ip

# Conflicts:
#	src/main/java/Bob.java

gekjunxu committed:
Change name to bob again

gekjunxu committed:
Added echo function

gekjunxu committed:
Added add and list feature

gekjunxu committed:
Add the ability to mark tasks as done, added Task class

gekjunxu committed:
Added deadline processing

gekjunxu committed:
Added todo, event support, level 4

gekjunxu committed:
Added pause so test does not auto quit after testing

gekjunxu committed:
added tests, and expected output

gekjunxu committed:
Added error handling, BobException class to handle invalid inputs

gekjunxu committed:
Moved list declaration outside try branch

gekjunxu committed:
Added delete function

gekjunxu committed:
update exception throwing to include detecting if argument is not a number

gekjunxu committed:
Update exception handling to include incorrect number of arguments

gekjunxu committed:
Correct spelling

gekjunxu committed:
Add feature to create datafile if not found

gekjunxu committed:
Change fomat of text block for exception to improve readability

gekjunxu committed:
Add feature to save data to file before program exits

gekjunxu committed:
Update event input parsing

gekjunxu committed:
Add space when printing end time of event

gekjunxu committed:
Add feature to save to and read tasks from disk

gekjunxu committed:
Merge branch 'branch-Level-7'

gekjunxu committed:
Add time and date parsing support for deadline tasks

gekjunxu committed:
Add date / time interpretation support for Event tasks

gekjunxu committed:
Merge branch 'branch-Level-8'

gekjunxu committed:
Add Ui class for more OOP

gekjunxu committed:
Add more error handling cases

gekjunxu committed:
Add testcases

gekjunxu committed:
Update script to point to correct file

gekjunxu committed:
Fix issue that causes test script to throw exception

gekjunxu committed:
Update test script to delete saved file before each run

gekjunxu committed:
Add parser and storage classes for more OOP

gekjunxu committed:
Add tasklist class

gekjunxu committed:
Merge branch 'branch-A-MoreOOP'

gekjunxu committed:
Add bob package

gekjunxu committed:
Add packages: storage, command, exception, task, ui

gekjunxu committed:
Update test script

gekjunxu committed:
Merge branch 'branch-A-Packages'

gekjunxu committed:
Merge remote-tracking branch 'origin/add-gradle-support'

gekjunxu committed:
Add natty date parser dependency

gekjunxu committed:
Add JUnit dependency

gekjunxu committed:
Add test for deadline and todo task

gekjunxu committed:
Add parsertest

gekjunxu committed:
Add test for parseDate

gekjunxu committed:
Edit formatting

gekjunxu committed:
Add some docs for parseTask

gekjunxu committed:
Add JavaDocs

gekjunxu committed:
Merge branch 'branch-A-CodingStandard'

gekjunxu committed:
Update formatting

gekjunxu committed:
Update formatting

gekjunxu committed:
Update listTasks

gekjunxu committed:
Add find feature

gekjunxu committed:
Update found function printout

gekjunxu committed:
Update test to include testing found function

gekjunxu committed:
Merge branch 'branch-Level-9'

gekjunxu committed:
Update formatting

gekjunxu committed:
Add checkstyle

gekjunxu committed:
Adjust indentations and formatting for checkstyle

gekjunxu committed:
Merge branch 'branch-A-CheckStyle'

gekjunxu committed:
Add JavaFX in gradle file

gekjunxu committed:
Increment to JavaFX tutorial part 2

gekjunxu committed:
Increment to JavaFX part 3 iteration 1

gekjunxu committed:
Increment to JavaFX Part 3 Iteration 2

gekjunxu committed:
Increment to JavaFX Part 3 iteration 3

johngao122 commented on src/main/java/bob/Bob.java:
```diff
@@ -214,7 +212,19 @@ public void run() throws IOException {
                         ui.printLine();
 
-                    } else {
+                    } else if (input[0].equalsIgnoreCase("find")) {
+                        if (input.length < 2) {
+                            throw new BobException("Incorrect number of arguments");
```
Have you considered using a switch statement instead of if else

johngao122 commented on src/main/java/bob/task/TaskList.java:
```diff
@@ -48,9 +48,26 @@ public void deleteTask(int index) {
+        ArrayList<Task> foundTasks = new ArrayList<>();
         for (Task task : tasks) {
-            System.out.println(task);
+            if (task.description.contains(name)) {
+                foundTasks.add(task);
+            }
```
Is direct field access intended here? Perhaps we should use a getter method to maintain encapsulation?

johngao122 commented on src/main/java/Bob.java:
```diff
@@ -190,4 +192,48 @@ public static void main(String[] args) throws BobException, IOException {
+
+
+    private static Task parseTask(String line) throws BobException {
+        char taskType = line.charAt(1); // The character after the first '['
```
Should validate line length before accessing characters.

johngao122 reviewed (commented):
I like the improvements to code organization and readability, particularly the extraction of parsing logic into a dedicated method. However, there are some concerns around error handling and input validation that should be addressed before merging.

gekjunxu committed:
Add most of the bot logic into GUI

The program closure logic when user types 'bye' and the initial file creation message for GUI is pending implementation

gekjunxu committed:
Implement datafile message and welcome message

gekjunxu committed:
Add feature to auto close when user types bye

gekjunxu committed:
Enable window resizing

gekjunxu committed:
Add CSS styling

gekjunxu committed:
Merge branch 'branch-Level10'

gekjunxu committed:
Change import order

gekjunxu committed:
Update gradle build name

gekjunxu committed:
Enable assertions

gekjunxu committed:
Add assertions

The code does not contain any assertions.

Assertions allow the runtime to detect assumptions about the program state, and potentially catch bugs.

Let's add assertions in constructors and methods.

gekjunxu committed:
Simplify printout message calling logic

The current implementation results in multiple duplicated code lines.

Let's change printAddedTaskMessage() in Ui class to take in the task toString
and the number of tasks in the list, we reduce the frequency where we call
printNumOfItemsInList directly, instead having printAddedTaskMessage method handle it.

gekjunxu committed:
Extract methods to improve readability

The current program logic is nested inside the switch block for each of the commands.

Let's extract out the methods for each command.

gekjunxu committed:
Remove unused dependencies

The natty dependency was originally included for future addition for enhanced date parsing.

However it is not planned for impementation anytime soon, let's remove unused dependencies in the meantime.

gekjunxu committed:
Extract methods for code quality

The bye command was originally not extracted.

To adhere to SLAP principle, let's abstract out that part of code into a new method as well.

gekjunxu committed:
Extract methods from parser

gekjunxu committed:
Add javadocs to extracted methods

gekjunxu committed:
Merge pull request #1 from gekjunxu/branch-A-Assertions

Merge Branch A-Assertions

gekjunxu committed:
Merge pull request #2 from gekjunxu/branch-A-CodeQuality

Branch a code quality

gekjunxu committed:
Set up CI

gekjunxu committed:
Replace dependency that was inadvertenly removed

gekjunxu committed:
Remove text ui tests

The text ui tests were suited for a CLI interface, and the chatbot has now been updated to a GUI version.

Let's remove the text-ui-test as it is no longer relevant.

gekjunxu committed:
Update gradle.yml

Remove redirection tests

gekjunxu committed:
Implement natty time parser for natural language parsing

gekjunxu committed:
Merge branch 'branch-BCD-Extension'

gekjunxu committed:
Set conditions to return null for invalid dates

gekjunxu committed:
Update ParserTest with more relevant tests

The use of natty allows for more date formats, and tries to parse dates with missing info,
and will not return null in many cases.

Let's update the test to include confirmed cases where natty would not be able to parse the date.

gekjunxu committed:
Clean up unused imports

gekjunxu committed:
Update indentation

gekjunxu committed:
Change boolean variable name to fit convention

gekjunxu committed:
Add credits for AI use

gekjunxu committed:
Improve Javadocs

gekjunxu committed:
Set program title

gekjunxu committed:
Add UI screenshot

gekjunxu committed:
Update README.md

gekjunxu committed:
Update README.md

gekjunxu committed:
Update README.md

gekjunxu committed:
Update README.md

gekjunxu committed:
Update README.md

gekjunxu committed:
Update README.md

gekjunxu committed:
Update README.md

gekjunxu committed:
Update README.md

gekjunxu committed:
Improve crop of Ui

gekjunxu committed:
Merge branch 'master' of https://github.com/gekjunxu/ip

gekjunxu committed:
Fix typo
